tests/filecheck/transforms/scf_for_loop_range_folding.mlir

```diff
@@ -0,0 +1,95 @@
+// RUN: xdsl-opt -p scf-for-loop-range-folding --split-input-file %s | filecheck %s
+
+func.func @fold_one_loop(%arg0: memref<?xi32>, %arg1: index, %arg2: index) {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  %c4 = arith.constant 4 : index
+  scf.for %i = %c0 to %arg1 step %c1 {
+    %0 = arith.addi %arg2, %i : index
+    %1 = arith.muli %0, %c4 : index
+    %2 = memref.load %arg0[%1] : memref<?xi32>
+    %3 = arith.muli %2, %2 : i32
+    memref.store %3, %arg0[%1] : memref<?xi32>
+  }
+  return
+}
+
+// CHECK:       %c0 = arith.constant 0 : index
+// CHECK-NEXT:  %c1 = arith.constant 1 : index
+// CHECK-NEXT:  %c4 = arith.constant 4 : index
+// CHECK-NEXT:  %0 = arith.addi %c0, %arg2 : index
+// CHECK-NEXT:  %1 = arith.addi %arg1, %arg2 : index
+// CHECK-NEXT:  %2 = arith.muli %0, %c4 : index
+// CHECK-NEXT:  %3 = arith.muli %1, %c4 : index
+// CHECK-NEXT:  %4 = arith.muli %c1, %c4 : index
+// CHECK-NEXT:  scf.for %i = %2 to %3 step %4 {
+// CHECK-NEXT:      %5 = memref.load %arg0[%i] : memref<?xi32>
+// CHECK-NEXT:      %6 = arith.muli %5, %5 : i32
+// CHECK-NEXT:      memref.store %6, %arg0[%i] : memref<?xi32>
+// CHECK-NEXT:  }
```

Comment:
I've been putting the check lines above the inputs recently, in conformance with the MLIR convention. Could you please make the same shift in this file? I know we're not really consistent here in xDSL